src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -58,42 +72,52 @@ def inner(func: TQuerySubscriptionCallable) -> TQuerySubscriptionCallable:
     return inner
 
 
-def parse_message_value(value: str) -> Dict[str, Any]:
+def parse_message_value(value: str, jsoncodec: JsonCodec) -> PayloadV3:
     """
     Parses the value received via the Kafka consumer and verifies that it
     matches the expected schema.
     :param value: A json formatted string
     :return: A dict with the parsed message
     """
+    old_version = False
+
     with metrics.timer("snuba_query_subscriber.parse_message_value.json_parse"):
-        wrapper: Dict[str, Any] = json.loads(value)
+        wrapper: SubscriptionResult = json.loads(value)
```

Comment:
Instead of calling `json.loads()` followed by `json_codec.validate()`, you can do `json_codec.decode(wrapper, validate=True)`. It uses rapidjson